src/item.cpp

```diff
@@ -3168,6 +3171,91 @@ int item::chip_resistance( bool worst ) const
     return res;
 }
 
+int item::damage() const {
+    return damage_ < 0 ? floor( damage_ ) : ceil( damage_ );
+}
+
+int item::min_damage() const
+{
+    return is_ammo() ? 0 : -1;
+}
+
+int item::max_damage() const
+{
+    return is_ammo() ? 0 : 4;
+}
+
+bool item::mod_damage( double qty, damage_type dt )
+{
+    bool destroy = false;
+
+    if( count_by_charges() ) {
+        charges -= std::min( type->stack_size * qty, double( charges ) );
+        destroy |= charges == 0;
+    }
+
+    if( qty > 0 ) {
+        on_damage( qty, dt );
+    }
+
+    destroy |= damage_ + qty > max_damage();
+
+    damage_ = std::max( std::min( damage_ + qty, double( max_damage() ) ), double( min_damage() ) );
+
+    return destroy;
+}
+
+nc_color item::damage_color() const
+{
+    // @todo unify with getDurabilityColor
+
+    // reinforced, undamaged and nearly destroyed items are special case
+    if( damage() < 0 ) {
+        return c_green;
+    }
+    if( damage() == 0 ) {
+        return c_ltgreen;
+    }
+    if( damage() == max_damage() ) {
+        return c_red;
+    }
+
+    // assign other colors proportionally
+    auto q = double( damage() ) / max_damage();
```

Comment:
Yes, see the `@todo` comment. I'm probably going to tackle that as part 2 with the symbol and colors being assigned based upon percentage.